### PR TITLE
[ONNX] Clean up asserts in libjit, and add an instruction verifier to check the group arg

### PR DESCRIPTION
**Pull request by beicy**

### CODE REVIEW

File: lib/IR/Instrs.cpp [hunk 1]
```diff
@@ -78,6 +78,28 @@ void AllocActivationInst::verify() const {
...
 }
 
+void ConvolutionInst::verify() const {
+  assert(getDest()->getElementType() == getSrc()->getElementType() &&
+         "Invalid Element Type");
+  assert(getDest()->getElementType() == getFilter()->getElementType() &&
+         "Invalid Element Type");
+  assert(getDest()->getElementType() == getBias()->getElementType() &&
+         "Invalid Element Type");
+  assert(getSrc()->dims()[3] % getGroup() == 0 &&
```
Comment (jfix71) [inline lib/IR/Instrs.cpp]: Are these checks not done in `ConvolutionNode::verify()`? Our general philosophy for verification for Nodes vs. Insts is to assume that verifying at the Node level is sufficient for these sorts of checks, especially for Nodes that are IRGen'd pretty directly into the Instr.

Comment (beicy) [reply lib/IR/Instrs.cpp]: @jfix71  ConvolutionNode checks it as well.  However,  currently, both of Convolution Node and Instruction checks if the types of dest, src, filter and bias are the same. For others, e.g SoftMax, the type check are done both in Node and Instr level.  So I added the check in instruction level as well.

Comment (jfix71) [reply lib/IR/Instrs.cpp]: @beicy We wanted to do the more "complicated" verification at the Node level, and try to have all the Instruction verification autogenerated. So the SoftMax instruction and node may verify the same things, but the SoftMaxInst has its verify autogenerated. If for example we wanted to verify something for SoftMax that is outside of the scope of autogenerated verify, it should be sufficient to just add it to the SoftMaxNode, and keep the SoftMaxInst the same/autogenerated.

So I think for this case it's probably safe to just remove the asserts from `libjit_conv.cpp` and leave the rest of the verification as is.

Comment (beicy) [reply lib/IR/Instrs.cpp]: @jfix71 Sure. I will work in this way. Thanks for your clarification:)

**Pull request merged**